### PR TITLE
1057 Add ignite handler for confusion matrix metric

### LINKED ISSUE #1057: Develop ignite handlers for new metrics
**Is your feature request related to a problem? Please describe.**
As @yiheng-wang-nv and @rijobro developed several new metrics recently, we need to develop the ignite handler wrappers for them.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Partly fix #1057 

### Description
This PR implemented the ignite handler for confusion matrix metric and fixed typo in `compute_confusion_matrix()`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/handlers/utils.py [hunk 1]
```diff
@@ -39,3 +42,13 @@ def stopping_fn(engine: Engine):
...
+
+
+def all_gather(tensor):
+    """
+    All gather the data of tensor value in distributed data parallel.
+    """
+    # create placeholder to collect the data from all processes
+    output = [torch.zeros_like(tensor) for _ in range(dist.get_world_size())]
+    dist.all_gather(output, tensor)
+    return torch.cat(output, dim=0)
```
Comment (wyli) [inline monai/handlers/utils.py:54]: will we have the case that some tensors are on gpu, some are not, what would be the device of the returned? should this utility check the property?

**Review by wyli — APPROVED** [review]

File: monai/handlers/utils.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/handlers/utils.py:54]: Hi @wyli ,

Thanks for your review.
I don't think we have that case, and `dist_all_gather()` should return an error for it.
I added a check for PyTorch distributed module status in the latest commit.

Thanks.
